src/main/java/com/codicesoftware/plugins/hudson/actions/CheckoutAction.java

```diff
@@ -108,31 +117,37 @@ private static void deleteWorkspace(
         workspaces.remove(workspace);
     }
 
-    @Deprecated
-    private static Workspace findWorkspaceByName(List<Workspace> workspaces, String workspaceName) {
+    protected static Workspace findWorkspaceByPath(FilePath workspacePath, List<Workspace> workspaces) {
         for (Workspace workspace : workspaces) {
-            if (workspace.getName().equals(workspaceName)) {
+            if (isSameWorkspacePath(workspace.getPath().getRemote(), workspacePath.getRemote())) {
                 return workspace;
             }
         }
         return null;
     }
 
-    private static Workspace findWorkspaceByPath(List<Workspace> workspaces, FilePath workspacePath) {
+    protected static List<Workspace> findWorkspacesInsidePath(FilePath workspacePath, List<Workspace> workspaces) {
+        String basePath = workspacePath.getRemote();
+
+        List<Workspace> result = new ArrayList<>();
         for (Workspace workspace : workspaces) {
-            if (isSameWorkspacePath(workspace.getPath().getRemote(), workspacePath.getRemote())) {
-                return workspace;
+            String testPath = workspace.getPath().getRemote();
+
+            if (isNestedWorkspacePath(basePath, testPath)) {
+                result.add(workspace);
             }
         }
-        return null;
+        return result;
     }
 
-    private static List<Workspace> findWorkspacesInsidePath(List<Workspace> workspaces, FilePath workspacePath) {
-        List<Workspace> result = new ArrayList<>();
+    protected static List<Workspace> findWorkspacesOutsidePath(FilePath workspacePath, List<Workspace> workspaces) {
```

Comment:
I think a better name for this would be `findParentWorkspaces()`, right?

Reply:
It's a matter of what is more understandable and have less ambiguity. Inner/Outer or Child/Parent. I used inner/outer and child/parent in the comments to ensure it would be understandable what this fragment does despite the disambiguity of the child/parent name. Would leave it but if you feel differently then maybe change it?

```
        // Handle situation where workspace exists in child path.
        List<Workspace> innerWorkspaces = findWorkspacesInsidePath(workspacePath, workspaces);
        for (Workspace workspace : innerWorkspaces) {
            deleteWorkspace(tool, workspace, workspaces);
        }

        // Handle situation where workspace exists in parent path.
        List<Workspace> outerWorkspaces = findWorkspacesOutsidePath(workspacePath, workspaces);
        for (Workspace workspace : outerWorkspaces) {
            deleteWorkspace(tool, workspace, workspaces);
        }
```

Reply:
We'll accept this name. We'll think if we can come up with a better naming 😉

Also, we're worried about the change from 'private' to 'protected' in this static class. We'll evaluate it and probably submit a new PR.

Reply:
So maybe `allChildWorkspaces` and `allParentWorkspaces` will be better?